### ISSUE TITLE
[Tiktok] Unable to download some Tiktok videos

### ISSUE THREAD

**Issue by triplesixman** [yt-dlp/yt-dlp#2243]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2021.12.27**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

US

### Description

Unable to download some Tiktok videos

### Verbose log

```shell
root@local:~/test# /usr/local/bin/yt-dlp.sh -Uv  https://www.tiktok.com/@chloecatalanooo/video/7041197121529203974?_t=8OnUindHzoj 
[debug] Command-line config: ['-Uv', 'https://www.tiktok.com/@chloecatalanooo/video/7041197121529203974?_t=8OnUindHzoj']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.12.27 [6223f67a8] (source)
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 1e5c83b26
[debug] Python version 3.9.7 (CPython 64bit) - Linux-4.9.0-17-amd64-x86_64-with-glibc2.24
[debug] exe versions: ffmpeg 4.1.2, ffprobe 4.1.2, phantomjs 2.1.1
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
Latest version: 2021.12.27, Current version: 2021.12.27
yt-dlp is up to date (2021.12.27)
[debug] [TikTok] Extracting URL: https://www.tiktok.com/@chloecatalanooo/video/7041197121529203974?_t=8OnUindHzoj
[TikTok] 7041197121529203974: Downloading video details
WARNING: [TikTok] 7041197121529203974: Failed to parse JSON  (caused by JSONDecodeError('Expecting value: line 1 column 1 (char 0)')); please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.; Retrying with webpage
[TikTok] 7041197121529203974: Downloading webpage
[TikTok] 7041197121529203974: Downloading video webpage
ERROR: [TikTok] 7041197121529203974: Unable to extract json_string; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
  File "/opt/yt-dlp/yt_dlp/extractor/common.py", line 599, in extract
    ie_result = self._real_extract(url)
  File "/opt/yt-dlp/yt_dlp/extractor/tiktok.py", line 368, in _real_extract
    json_string = self._search_regex(
  File "/opt/yt-dlp/yt_dlp/extractor/common.py", line 1173, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
```

**Comment by artiemog** [comment]
I've had this issue as well. In my non-exhaustive testing, it seems like `vm.tiktok.com` links are handled fine, but those same videos cannot be downloaded if the URL is given in the format of `tiktok.com/@[username]/video/[video-ID]`

**Comment by dmattera** [comment]
getting the exact same error on every link I've tried this morning (~10) using same version of yt-dlp 2021.12.27

**Comment by triplesixman** [comment]
> I've had this issue as well. In my non-exhaustive testing, it seems like `vm.tiktok.com` links are handled fine, but those same videos cannot be downloaded if the URL is given in the format of `tiktok.com/@[username]/video/[video-ID]`

A question that has nothing to do with the problem, how are the "vm.tiktok.com" links generated? So far I've only had "tiktok.com" links.

**Comment by dmattera** [comment]
it looks like the `__NEXT_DATA__`  script that `extractor/tiktok.py` uses to extract the `json_string` is no longer being included in the HTML sent by tiktok.

<img width="1168" alt="Screen Shot 2022-01-06 at 12 50 25 PM" src="https://user-images.githubusercontent.com/13613522/148428971-49d7821d-83cb-4425-b23a-a56e04ae961e.png">

The whole desktop version of site looks like it's been updated as well so I'm guess they probably pushed out some big rollout last night.

**Comment by artiemog** [comment]
> > I've had this issue as well. In my non-exhaustive testing, it seems like `vm.tiktok.com` links are handled fine, but those same videos cannot be downloaded if the URL is given in the format of `tiktok.com/@[username]/video/[video-ID]`
> 
> A question that has nothing to do with the problem, how are the "vm.tiktok.com" links generated? So far I've only had "tiktok.com" links.

I don't use the site directly, I only watch videos that I'm sent, but I believe they're generated by the share button in the phone app.

**Comment by dmattera** [comment]
From what I can tell, the JSON data needed is now in a `script` element with an id = `sigi-persisted-data`

it looks like all the data yt-dlp extracts has been moved around/renamed. for example the `downloadAddr` which was located here in the json_data:

`json_data['props']['pageProps']['itemInfo']['itemStruct']['video']['downloadAddr']`

now appears to be located in two places, here:

`json_data['ItemModule'][<video_id>]['video']['downloadAddr']`

and here:

`json_data['ItemList']['video']['preloadList'][0]['url']`

I would be willing to give a go at updating `extractor/tiktok.py` to reflect these changes but this would be my first time working on anything related to yt-dlp and so I would probably need some guidance, especially on updating the regEx piece for extracting the `json_string` as I'm pretty useless when it comes to regEx in general.

**Comment by dirkf** [comment]
See also #2133.

The fix for SIGI-type pages appears to be included in the yt-dlp version that gets run as `Git HEAD: f5c2c2c9b`, and was observed and patched for yt-dl some time ago as well: PR #[302224](https://github.com/ytdl-org/youtube-dl/pull/30224), issue #[30251](https://github.com/ytdl-org/youtube-dl/issues/30251). There have been several commits to the extractor since the release, including [the SIGI fix](https://github.com/yt-dlp/yt-dlp/pull/2164/commits/8ef69868e7f3d9d745dabff04e9a8489c48e049a).

```
$ python3.9 -m yt_dlp -v -F 'https://www.tiktok.com/@chloecatalanooo/video/7041197121529203974?_t=8OnUindHzoj'
[debug] Command-line config: ['-v', '-F', 'https://www.tiktok.com/@chloecatalanooo/video/7041197121529203974?_t=8OnUindHzoj']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.12.27 [6223f67a8] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: f5c2c2c9b
[debug] Python version 3.9.7 (CPython 32bit) - Linux-4.4.0-210-generic-i686-with-glibc2.23
[debug] exe versions: avconv 4.3, avprobe 4.3, ffmpeg 4.3, ffprobe 4.3
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
[debug] [TikTok] Extracting URL: https://www.tiktok.com/@chloecatalanooo/video/7041197121529203974?_t=8OnUindHzoj
[TikTok] 7041197121529203974: Downloading video details
[debug] Sort order given by extractor: quality, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, hdr:12(7), asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 7041197121529203974:
ID                    EXT RESOLUTION │    FILESIZE   TBR PROTO │ VCODEC   VBR ACODEC ABR MORE INFO
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
download_addr-0       mp4 576x1024   │     2.33MiB       https │ h264         aac        Download video, watermarked (API)
download_addr-1       mp4 576x1024   │     2.33MiB       https │ h264         aac        Download video, watermarked
download_addr-2       mp4 576x1024   │     2.33MiB       https │ h264         aac        Download video, watermarked
h264_540p_1888257-0   mp4 576x1024   │     2.30MiB 1888k https │ h264   1888k aac     0k Direct video (API)
h264_540p_1888257-1   mp4 576x1024   │     2.30MiB 1888k https │ h264   1888k aac     0k Direct video
h264_540p_1888257-2   mp4 576x1024   │     2.30MiB 1888k https │ h264   1888k aac     0k Direct video
bytevc1_540p_310191-0 mp4 576x1024   │   387.66KiB  310k https │ h265    310k aac     0k Playback video (API)
bytevc1_540p_310191-1 mp4 576x1024   │   387.66KiB  310k https │ h265    310k aac     0k Playback video
bytevc1_540p_310191-2 mp4 576x1024   │   387.66KiB  310k https │ h265    310k aac     0k Playback video
bytevc1_540p_407699-0 mp4 576x1024   │   509.53KiB  407k https │ h265    407k aac     0k Playback video (API)
bytevc1_540p_407699-1 mp4 576x1024   │   509.53KiB  407k https │ h265    407k aac     0k Playback video
bytevc1_540p_407699-2 mp4 576x1024   │   509.53KiB  407k https │ h265    407k aac     0k Playback video
bytevc1_540p_689966-0 mp4 576x1024   │   862.29KiB  689k https │ h265    689k aac     0k Direct video (API)
bytevc1_540p_689966-1 mp4 576x1024   │   862.29KiB  689k https │ h265    689k aac     0k Direct video
bytevc1_540p_689966-2 mp4 576x1024   │   862.29KiB  689k https │ h265    689k aac     0k Direct video
bytevc1_720p_806576-0 mp4 unknown    │  1008.02KiB  806k https │ h265    806k aac     0k Playback video (API)
bytevc1_720p_806576-1 mp4 unknown    │  1008.02KiB  806k https │ h265    806k aac     0k Playback video
bytevc1_720p_806576-2 mp4 unknown    │  1008.02KiB  806k https │ h265    806k aac     0k Playback video
$
``` 
Whereas in yt-dl, the extractor tries to get a cookie for the page of interest by fetching `https://www.tiktok.com/` and this now times out in all Python versions tested.

**Comment by coletdjnz** [comment]
Yep, duplicate of #2133. Fix is already in master and will be in next release.
